core/src/runtime/service.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <libp2p/log/logger.hpp>
+
+#include <wasm-features.h>
+#include <shell-interface.h>
+
+#include "chain/spec.h"
+#include "network/peer_manager.h"
+#include "runner/client_runner.h"
+#include "runtime/api.h"
+#include "runtime/executor.h"
+#include "runtime/module.h"
+
+namespace plc::core::runtime {
+
+class Service final {
+public:
+    enum class Error {
+        DecompresionError = 1,
+        ObtainingRuntimeError,
+        MissingInitialRuntime,
+    };
+
+    Service(std::shared_ptr<plc::core::chain::Spec> spec,
+            std::shared_ptr<plc::core::network::PeerManager> connection_manager, 
+            std::shared_ptr<plc::core::runner::ClientRunner> runner) : 
+            m_spec(spec), m_connection_manager(connection_manager), m_runner(runner) {
+                m_module = std::make_shared<Module>();
+                m_executor = std::make_shared<Executor>();
+                m_api = std::make_shared<Api>(m_executor);
+            }
+
+    Result<void> loadGenesisRuntime();
```

Comment:
Just my personal opinion. I would rather make all the services active - make them execute everything they need on their own. E.g. for this service:
- When created it calls a coroutine on peer manager that returns peer when connection is established
- Then calls coroutine to get the runtime for the current state
- Subscribes to new block events and tracks if the runtime has changed

So that the service will be able to return the Runtime for the current block

Reply:
This is initial the implementation - I'll think about your idea in the next PR I'm currently working on.